packages/hash/api/src/model/knowledge/workspaceInstance.model.ts

```diff
@@ -0,0 +1,180 @@
+  UserModel,
+  LinkModel,
+} from "..";
+import { workspaceAccountId } from "../util";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+import { EntityTypeMismatchError, NotFoundError } from "../../lib/error";
+
+export type WorkspaceInstanceModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel" | "ownedById"
+> & {};
+
+/**
+ * @class {@link WorkspaceInstanceModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): WorkspaceInstanceModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.workspaceInstance.schema.$id
+    ) {
+      throw new EntityTypeMismatchError(
+        entity.entityId,
+        WORKSPACE_TYPES.entityType.workspaceInstance.schema.$id,
+        entity.entityTypeModel.schema.$id,
+      );
+    }
+
+    return new WorkspaceInstanceModel(entity);
+  }
+
+  /**
+   * Create the workspace instance entity.
+   *
+   * @see {@link EntityModel.create} for the remaining params
+   */
+  static async createWorkspaceInstance(
+    graphApi: GraphApi,
+    params: WorkspaceInstanceModelCreateParams,
+  ) {
+    // Ensure the workspace instance entity has not already been created.
+    await WorkspaceInstanceModel.getWorkspaceInstanceModel(graphApi)
+      .then(() => {
+        throw new Error("Workspace instance entity already exists.");
+      })
+      .catch((error: Error) => {
+        if (!(error instanceof NotFoundError)) {
+          throw error;
+        }
+      });
+
+    const { actorId } = params;
+
+    const entityTypeModel = WORKSPACE_TYPES.entityType.workspaceInstance;
+
+    const entityModel = await EntityModel.create(graphApi, {
+      ownedById: workspaceAccountId,
+      properties: {},
+      entityTypeModel,
+      actorId,
+    });
+
+    return WorkspaceInstanceModel.fromEntityModel(entityModel);
+  }
+
+  /**
+   * Get the workspace instance.
+   */
+  static async getWorkspaceInstanceModel(
+    graphApi: GraphApi,
+  ): Promise<WorkspaceInstanceModel> {
+    const entities = await EntityModel.getByQuery(graphApi, {
+      all: [
+        { equal: [{ path: ["version"] }, { parameter: "latest" }] },
+        {
+          equal: [
+            { path: ["type", "versionedUri"] },
+            {
+              parameter:
+                WORKSPACE_TYPES.entityType.workspaceInstance.schema.$id,
+            },
+          ],
+        },
+      ],
+    });
+
+    if (entities.length > 1) {
+      throw new Error(
+        "More than one workspace instance entity found in the graph.",
+      );
+    }
+
+    const entity = entities[0];
+
+    if (!entity) {
+      throw new NotFoundError("Could not find workspace instance entity.");
+    }
+
+    return WorkspaceInstanceModel.fromEntityModel(entity);
+  }
+
+  /**
+   * Add an instance admin to the workspace instance.
+   *
+   * @param params.userModel - the user that may be a workspace instance admin.
+   */
+  async hasAdmin(
```

Comment:
I think this name is confusing me slightly, as it implies to me that there is only a single instance admin. Perhaps we should call it `isUserAdmin` or something?

Reply:
Hmm, maybe it's just me but `HashInstanceModel.hasAdmin({ userModel })` makes more sense to me than `HashInstanceModel.isAdmin({ userModel })` when I read it in my head.

Would you feel better about `HashInstanceModel.hasUserAdmin({ userModel })`? It just seems somewhat superfluous because only users can be admins.

Reply:
I agree that `hasUserAdmin` would seem a little superfluous. And I see how the method call sort of looks like `subject predicate object`. I guess looking at the method in the class is what made me think of `isUserAdmin` instead. I don't think we should change this :+1: